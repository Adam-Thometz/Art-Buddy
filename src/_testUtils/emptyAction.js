const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION