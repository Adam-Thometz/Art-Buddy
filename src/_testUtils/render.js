import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
