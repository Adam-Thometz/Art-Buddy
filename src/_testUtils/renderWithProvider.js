import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;