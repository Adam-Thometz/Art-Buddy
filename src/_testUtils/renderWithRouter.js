import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;