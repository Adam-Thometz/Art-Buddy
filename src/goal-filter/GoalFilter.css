.GoalFilter {
  width: 80%;
  text-align: left;
  font-size: 2rem;
  margin: 0 auto;
}

.GoalFilter-list {
  list-style-type: none;
}

.GoalFilter-list-item {
  display: flex;
  align-items: center;
  padding: 1rem 0;
  color: var(--ctblue, #4266B4);
  text-decoration-line: underline;
  cursor: pointer;
}

.GoalFilter-icon {
  width: 3rem;
  height: 3rem;
  padding-right: 1rem;
}