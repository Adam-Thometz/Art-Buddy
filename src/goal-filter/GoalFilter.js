import { useContext } from "react";
import { PopupContext } from "_context/PopupContext";

import "./GoalFilter.css";

import GoalCategory from "./category/GoalCategory";

import { goalIcons } from "_media/goal-filter/goalFilter.assets";

const GoalFilter = () => {
  const { setCurrPopup } = useContext(PopupContext);

  const capitalize = word => word[0].toUpperCase() + word.slice(1);

  const showFilteredGames = e => {
    const { id: category } = e.currentTarget;
    const title = <span>
      <img className="GoalFilter-icon" src={goalIcons[category]} alt="" />
      <span>{capitalize(category)} Standards</span>
    </span>
    setCurrPopup({
      title,
      popup: <GoalCategory category={category} />
    });
  }
  return (
    <div className="GoalFilter">
      <p className="GoalFilter-description">
        Art Buddy was designed to seamlessly integrate the Common Core Standards!
        Look for the icon(s) on each game!
      </p>
      <ul className="GoalFilter-list">
        {Object.keys(goalIcons).map(goal => (
          <li key={goal} id={goal} className="GoalFilter-list-item" onClick={showFilteredGames}>
            <img className="GoalFilter-icon" src={goalIcons[goal]} alt="" />
            <span>{capitalize(goal)} Standards</span>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default GoalFilter;
