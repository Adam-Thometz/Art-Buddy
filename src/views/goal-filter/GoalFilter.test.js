import GoalFilter from "./GoalFilter";

import { render } from "_testUtils/render";

describe("GoalFilter component", () => {
  it("renders without crashing", () => {
    render(<GoalFilter />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<GoalFilter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
