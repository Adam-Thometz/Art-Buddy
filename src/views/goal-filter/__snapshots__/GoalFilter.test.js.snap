// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalFilter component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GoalFilter"
  >
    <p
      class="GoalFilter-description"
    >
      Art Buddy was designed to seamlessly integrate the Common Core Standards! Look for the icon(s) on each game!
    </p>
    <ul
      class="GoalFilter-list"
    >
      <li
        class="GoalFilter-list-item"
        id="reading"
      >
        <img
          alt=""
          class="GoalFilter-icon"
          src="file"
        />
        <span>
          Reading Standards
        </span>
      </li>
      <li
        class="GoalFilter-list-item"
        id="writing"
      >
        <img
          alt=""
          class="GoalFilter-icon"
          src="file"
        />
        <span>
          Writing Standards
        </span>
      </li>
      <li
        class="GoalFilter-list-item"
        id="math"
      >
        <img
          alt=""
          class="GoalFilter-icon"
          src="file"
        />
        <span>
          Math Standards
        </span>
      </li>
      <li
        class="GoalFilter-list-item"
        id="communication"
      >
        <img
          alt=""
          class="GoalFilter-icon"
          src="file"
        />
        <span>
          Communication Standards
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
